programs/marinade-finance/src/instructions/liq_pool/liquid_unstake.rs

```diff
@@ -86,22 +86,25 @@ impl<'info> LiquidUnstake<'info> {
     // fn liquid_unstake()
     pub fn process(&mut self, msol_amount: u64) -> Result<()> {
         self.check_get_msol_from(msol_amount)?;
-        let is_treasury_msol_ready_for_transfer = self
+        let user_sol_balance = self.transfer_sol_to.lamports();
+        let user_msol_balance = self.get_msol_from.amount;
+        let treasury_msol_balance = self
             .state
-            .check_treasury_msol_account(&self.treasury_msol_account)?;
```

Comment:
Rename a function to treasury_msol_balance. Otherwise, it is doing unexpected thing

Reply:
renamed to `get_treasury_msol_balance`